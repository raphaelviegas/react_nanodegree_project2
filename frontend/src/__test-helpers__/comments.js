const uuidv4 = require('uuid/v4')

export const initialState = { comments: [] }

export const comments = [
    {
        id: '894tuq4ut84ut8v4t8wun89g',
        parentId: "8xf0y6ziyjabvozdd253nd",
        timestamp: 1468166872634,
        body: 'Hi there! I am a COMMENT.',
        author: 'thingtwo',
        voteScore: 6,
        deleted: false,
        parentDeleted: false
    },
    {
        id: '8tu4bsun805n8un48ve89',
        parentId: "8xf0y6ziyjabvozdd253nd",
        timestamp: 1469479767190,
        body: 'Comments. Are. Cool.',
        author: 'thingone',
        voteScore: -5,
        deleted: false,
        parentDeleted: false
    }
]

export const getCommentsId = '894tuq4ut84ut8v4t8wun89g'

export const removeComment = '894tuq4ut84ut8v4t8wun89g'

export const voteComment = {
    id: '894tuq4ut84ut8v4t8wun89g',
    voteScore: 10
}

let id = uuidv4()
export const addCommentInfo = {
    parentId: '8xf0y6ziyjabvozdd253nd',
    author: 'Test Author',
    body: 'Test Body',
    id
}

export const editCommentInfo = {
    id: '894tuq4ut84ut8v4t8wun89g',
    editContent: {
        body: 'Edited comment body',
        timestamp: Date.now()
    }
}