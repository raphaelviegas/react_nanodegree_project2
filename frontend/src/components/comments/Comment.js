import React, { Component } from 'react'
import PropTypes from 'prop-types'
import Moment from 'react-moment';
import VoteScore from '../shared/VoteScore'

class Comment extends Component {
    static propTypes = {
        commentInfo: PropTypes.object.isRequired,
        onRemoveComment: PropTypes.func.isRequired,
        onEditComment: PropTypes.func.isRequired,
    }

    removeComment(id) {
        this.props.onRemoveComment(id)
    }

    editComment(id, body) {
        this.props.onEditComment(id, body)
    }

    render() {
        const { author, timestamp, body, voteScore, id } = this.props.commentInfo
        return (
            <li>
                <div className="card">
                    <div className="card-content">
                        <span className="card-title comment-title">Author: {author}</span>
                        <span className="card-date comment-date"><Moment format="DD/MMM/YY">{timestamp}</Moment></span>
                        <p className="card-body comment-body">{body}</p>
                    </div>
                    <div className="card-action comment-action">
                        <span className="comment-action-edit" onClick={() => this.editComment(id, body)}>Edit</span>
                        <span className="comment-action-remove" onClick={() => this.removeComment(id)}>Delete</span>
                    </div>
                    <div className="comment-vote-score">
                        <VoteScore size={12} score={voteScore} type='comment' id={id} />
                    </div>
                </div>
            </li>
        )
    }
}

export default Comment
