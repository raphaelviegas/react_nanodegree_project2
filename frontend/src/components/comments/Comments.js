import React, { Component } from 'react'
import { connect } from 'react-redux'
import { withRouter } from 'react-router-dom'
import PropTypes from 'prop-types'
import { fetchComments, fetchRemoveComment } from '../../actions/comments'
import MaterialIcon from 'material-icons-react'
import sortBy from 'sort-by'
import AddCommentButton from './addComment/AddCommentButton'
import EditCommentModal from './editComment/EditCommentModal'
import Comment from './Comment'
import M from "materialize-css/dist/js/materialize.min.js"

export class Comments extends Component {
    static propTypes = {
        parentId: PropTypes.string.isRequired,
        comments: PropTypes.array.isRequired
    }

    state = {
        sortByOption: '-voteScore',
        commentId: '',
        commentBody: '',
        editModalOpen: ''
    }

    showEditModal = (id, body) => this.setState(
        () => ({ editModalOpen: 'modal-open', commentId: id, commentBody: body })
    )

    hideEditModal = () => this.setState(
        () => ({ editModalOpen: '', commentId: '', commentBody: '', })
    )

    componentDidMount() {
        this.props.fetchComments(this.props.parentId)
        let elems = document.querySelectorAll('.dropdown-trigger');
        // eslint-disable-next-line
        let instances = M.Dropdown.init(elems, { alignment: 'right', constrainWidth: false, hover: false });
    }

    removeComment = (id) => {
        this.props.fetchRemoveComment(id)
    }

    changeSortCriteria = (option) => {
        switch (option) {
            case 'voteScore':
                this.setState({ sortByOption: 'voteScore' })
                break
            case '-voteScore':
                this.setState({ sortByOption: '-voteScore' })
                break
            case 'timestamp':
                this.setState({ sortByOption: 'timestamp' })
                break
            case '-timestamp':
                this.setState({ sortByOption: '-timestamp' })
                break
            default:
                break
        }
    }

    render() {
        const { comments, parentId } = this.props
        const commentCount = comments.length
        const { sortByOption, commentId, commentBody, editModalOpen } = this.state
        //Sort By
        comments.sort(sortBy(sortByOption))
        return (
            <div className="post-comments">
                <div className="post-comments-title">
                    Comments: {commentCount}
                    <div className="order-by">
                        <a className='dropdown-trigger' data-target='dropdown1'><MaterialIcon icon="sort" size={22} /></a>
                        <ul id='dropdown1' className='dropdown-content'>
                            <li><span>Vote Score</span></li>
                            <li><a className="order-button" onClick={() => this.changeSortCriteria('voteScore')}>Ascending</a></li>
                            <li><a className="order-button" onClick={() => this.changeSortCriteria('-voteScore')}>Descending</a></li>
                            <li className="divider" tabIndex="-1"></li>
                            <li><span>Date</span></li>
                            <li><a className="order-button" onClick={() => this.changeSortCriteria('timestamp')}>Ascending</a></li>
                            <li><a className="order-button" onClick={() => this.changeSortCriteria('-timestamp')}>Descending</a></li>
                        </ul>
                    </div>
                </div>
                <div className="divider"></div>
                {commentCount === 0
                    ? <div className="noComments"><h6>No comments to show</h6></div>
                    : <ul className="comments">
                        {comments.map(comment => (
                            <Comment
                                commentInfo={comment}
                                key={comment.id}
                                onRemoveComment={this.removeComment}
                                onEditComment={this.showEditModal}
                            />
                        ))}
                    </ul>
                }
                <AddCommentButton parentId={parentId} />
                <EditCommentModal
                    modalOpen={editModalOpen}
                    commentId={commentId}
                    commentBody={commentBody}
                    onCloseModal={this.hideEditModal}
                />
            </div>
        )
    }
}

const mapStateToProps = (state, ownProps) => {
    return {
        parentId: ownProps.parentId,
        comments: state.commentsReducer.comments
    }
}

const mapDispatchToProps = (dispatch) => {
    return {
        fetchComments: (parentId) => dispatch(fetchComments(parentId)),
        fetchRemoveComment: (id) => dispatch(fetchRemoveComment(id))
    }
}

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(Comments));