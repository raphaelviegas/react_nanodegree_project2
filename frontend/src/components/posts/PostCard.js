import React, { Component } from 'react'
import { Link } from 'react-router-dom'
import PropTypes from 'prop-types'
import MaterialIcon from 'material-icons-react'
import VoteScore from '../shared/VoteScore'
import EditPostForm from '../posts/editPost/EditPostForm'

class PostCard extends Component {
    static propTypes = {
        post: PropTypes.object.isRequired,
        onRemovePost: PropTypes.func.isRequired
    }

    state = {
        showEdit: '',
    }

    showEdit = () => this.setState(() => ({ showEdit: 'card-reveal-active' }))
    hideEdit = () => this.setState(() => ({ showEdit: '' }))

    removePost(id) {
        this.props.onRemovePost(id)
    }
    render() {
        const { id, title, body, author, voteScore, commentCount, category } = this.props.post
        const { showEdit } = this.state
        return (
            <div className="col s12 m6">
                <div className="card grey lighten-4">
                    <div className="card-content">
                        <span className="card-title truncate">{title}</span>
                        <span className="card-author">{author}</span>
                        <p>{body}</p>
                    </div>
                    <div className="card-action valign-wrapper">
                        <VoteScore size={16} score={voteScore} type='post' id={id} />
                        <Link className="align-right" to={`/${category}/${id}`}>Show Post</Link>
                    </div>
                    <div className="card-comments">
                        <span className="card-comments-number">{commentCount}</span>
                    </div>
                    <div className="card-buttons">
                        <div className="card-button-edit" onClick={() => this.showEdit()}>
                            <MaterialIcon icon="edit" size={14} />
                        </div>
                        <div className="card-button-delete" onClick={() => this.removePost(id)}>
                            <MaterialIcon icon="delete" size={14} />
                        </div>
                    </div>
                    <div className={`card-reveal ${showEdit}`}>
                        <EditPostForm
                            id={id}
                            defaultTitle={title}
                            defaultBody={body}
                            onPostCardFormSubmited={this.hideEdit}
                        />
                        <div className="hide-edit" onClick={() => this.hideEdit()}>
                            <MaterialIcon icon="close" color="#c3c3c3" />
                        </div>
                    </div>
                </div>
            </div>
        )
    }
}

export default PostCard
