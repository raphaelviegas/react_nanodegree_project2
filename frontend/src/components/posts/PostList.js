import React, { Component } from 'react'
import { connect } from 'react-redux'
import { withRouter } from 'react-router-dom'
import PropTypes from 'prop-types'
import { fetchRemovePost } from '../../actions/posts'
import PostCard from './PostCard'
import sortBy from 'sort-by'

export class PostList extends Component {
  static propTypes = {
    posts: PropTypes.array.isRequired,
    sortByOption: PropTypes.string.isRequired
  }

  removePost = (id) => {
    this.props.fetchRemovePost(id)
  }

  render() {
    const { posts, sortByOption } = this.props
    //SortBy
    const sortedPosts = [...posts].sort(sortBy(sortByOption))
    return (
      <div>
        {posts.length === 0
          ? <div className="noPosts"><h4>No posts to show</h4></div>
          : <div className="row">
            {
              sortedPosts.map((post, index) => {
                return <PostCard post={post} onRemovePost={this.removePost} key={index} />
              })
            }
          </div>
        }
      </div>
    )
  }
}

const mapStateToProps = (state, ownProps) => {
  return {
    posts: ownProps.posts,
    sortByOption: state.postsReducer.sortByOption
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    fetchRemovePost: (id) => dispatch(fetchRemovePost(id))
  }
}

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(PostList));
