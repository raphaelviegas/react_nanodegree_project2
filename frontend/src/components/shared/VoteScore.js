import React, { Component } from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'
import MaterialIcon from 'material-icons-react'
import { fetchVotePost } from '../../actions/posts'
import { fetchVoteComment } from '../../actions/comments'

export class VoteScore extends Component {
    static propTypes = {
        size: PropTypes.number.isRequired,
        id: PropTypes.string,
        score: PropTypes.number.isRequired,
        type: PropTypes.string.isRequired
    }

    vote({ type, id, option }) {
        switch (type) {
            case 'post':
                this.props.fetchVotePost(id, { option })
                break
            case 'comment':
                this.props.fetchVoteComment(id, { option })
                break
            default:
                break
        }
    }

    render() {
        const { size, id, score, type } = this.props
        return (
            <div className="vote-score">
                <div className="vote-score-button vote-down" onClick={() => this.vote({ type, id, option: 'downVote' })}>
                    <MaterialIcon icon="thumb_down" size={size} />
                </div>
                <span className="vote-score-score">{score}</span>
                <div className="vote-score-button vote-up" onClick={() => this.vote({ type, id, option: 'upVote' })}>
                    <MaterialIcon icon="thumb_up" size={size} />
                </div>
            </div>
        )
    }
}

const mapStateToProps = (state, ownProps) => {
    return {
        size: ownProps.size,
        id: ownProps.id,
        score: ownProps.score,
        type: ownProps.type
    }
}

const mapDispatchToProps = (dispatch) => {
    return {
        fetchVotePost: (id, option) => dispatch(fetchVotePost(id, option)),
        fetchVoteComment: (id, option) => dispatch(fetchVoteComment(id, option))
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(VoteScore);
