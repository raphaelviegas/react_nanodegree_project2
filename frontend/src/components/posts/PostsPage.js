import React, { Component } from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'
import { fetchPosts } from '../../actions/posts'
import PostList from '../posts/PostList'
import OrderPosts from '../shared/OrderPosts'

export class PostsPage extends Component {
    static propTypes = {
        posts: PropTypes.array.isRequired
    }

    componentDidMount() {
        this.props.fetchPosts()
        let elems = document.querySelectorAll('.dropdown-trigger');
        // eslint-disable-next-line
        let instances = M.Dropdown.init(elems, { alignment: 'right', constrainWidth: false })
    }

    render() {
        return (
            <div>
                <div className="category-title">
                    Showing All Posts
                <OrderPosts />
                </div>
                <div className="divider"></div>
                <PostList posts={this.props.posts} />
            </div>
        )
    }
}

const mapStateToProps = (state) => {
    return {
        posts: state.postsReducer.posts,
    }
}

const mapDispatchToProps = (dispatch) => {
    return {
        fetchPosts: () => dispatch(fetchPosts())
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(PostsPage)
