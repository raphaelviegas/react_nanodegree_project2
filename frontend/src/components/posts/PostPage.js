import React, { Component } from 'react'
import { connect } from 'react-redux'
import { withRouter, Redirect } from 'react-router-dom'
import { fetchPosts, fetchRemovePost } from '../../actions/posts'
import MaterialIcon from 'material-icons-react'
import Moment from 'react-moment';
import VoteScore from '../shared/VoteScore'
import Comments from '../comments/Comments'
import EditPostModal from '../posts/editPost/EditPostModal'

export class PostPage extends Component {
    state = {
        modalOpen: ''
    }

    openModal = () => this.setState(() => ({ modalOpen: 'modal-open' }))
    closeModal = () => this.setState(() => ({ modalOpen: '' }))

    componentDidMount() {
        this.props.fetchPosts()
    }

    componentDidUpdate(prevProps) {
        if (this.props.id !== prevProps.id) {
            this.props.fetchPosts()
        }
    }

    removePost = (id) => {
        this.props.fetchRemovePost(id)
        this.props.history.push('/')
    }

    render() {
        const { postInfo } = this.props
        let invalidPost = (postInfo.length > 0) ? false : true
        return (
            <div>
                {invalidPost
                    ? <Redirect to="/error" />
                    : <div className="post-page">
                        {postInfo.map((post) => (
                            <div key={post.id}>
                                <div className="post-buttons">
                                    <div className="post-button-edit" onClick={() => this.openModal()}>
                                        <MaterialIcon icon="edit" size={20} />
                                    </div>
                                    <div className="post-button-remove" onClick={() => this.removePost(post.id)}>
                                        <MaterialIcon icon="delete" size={20} />
                                    </div>
                                </div>
                                <div className="container">
                                    <h4 className="post-title">{post.title}</h4>
                                    <div className="post-infos">
                                        <span className="post-author">Author: {post.author}</span>
                                        <span className="post-category">Category: {post.category}</span>
                                        <span className="post-date">
                                            <MaterialIcon icon="calendar_today" size={16} />
                                            <Moment format="DD/MMM/YY">{post.timestamp}</Moment>
                                        </span>
                                    </div>
                                    <VoteScore size={18} score={post.voteScore} type='post' id={post.id} />
                                    <p className="post-body">{post.body}</p>
                                    <Comments parentId={post.id} />
                                </div>
                                <EditPostModal
                                    modalOpen={this.state.modalOpen}
                                    onCloseModal={this.closeModal}
                                    id={post.id}
                                    title={post.title}
                                    body={post.body}
                                />
                            </div>
                        ))}
                    </div>
                }
            </div>
        )
    }
}
const mapStateToProps = (state, ownProps) => {
    return {
        id: ownProps.match.params.postId,
        postInfo: state.postsReducer.posts.filter(post => post.id === ownProps.match.params.postId)
    }
}

const mapDispatchToProps = (dispatch) => {
    return {
        fetchPosts: () => dispatch(fetchPosts()),
        fetchRemovePost: (id) => dispatch(fetchRemovePost(id))
    }
}

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(PostPage));
