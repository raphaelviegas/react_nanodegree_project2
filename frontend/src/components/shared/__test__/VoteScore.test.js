import React from 'react'
import { shallow } from "enzyme"
import { VoteScore } from '../VoteScore'
import MaterialIcon from 'material-icons-react'
import { votePost } from '../../../__test-helpers__/posts'
import { voteComment } from '../../../__test-helpers__/comments'

describe("<VoteScore />", () => {
    // mock function to replace the one provided by mapDispatchToProps
    const setupPost = {
        id: votePost.id,
        score: votePost.voteScore,
        type: 'post',
        size: 30,
        fetchVotePost: jest.fn()
    }
    const setupComment = {
        id: voteComment.id,
        score: voteComment.voteScore,
        type: 'comment',
        size: 20,
        fetchVoteComment: jest.fn()
    }

    beforeEach(() => {
        setupPost.fetchVotePost.mockClear()
        setupComment.fetchVoteComment.mockClear()
    })

    it('Should render two buttons', () => {
        const wrapper = shallow(<VoteScore {...setupPost} />)
        expect(wrapper.find('.vote-score-button').length).toEqual(2)
    })

    it('Should render a .vote-down button', () => {
        const wrapper = shallow(<VoteScore {...setupPost} />)
        expect(wrapper.find('.vote-down').length).toEqual(1)
    })

    it('Should render a .vote-up button', () => {
        const wrapper = shallow(<VoteScore {...setupPost} />)
        expect(wrapper.find('.vote-up').length).toEqual(1)
    })

    describe("<VoteScore  type='posts' />", () => {
        const wrapper = shallow(<VoteScore {...setupPost} />)

        it('Should render corretly on type Post', () => {
            expect(wrapper).toMatchSnapshot()
        })

        it('Should call fetchVotePost when the buttons are clicked', () => {
            expect(setupPost.fetchVotePost).toHaveBeenCalledTimes(0)
            wrapper.find('.vote-down').simulate('click')
            expect(setupPost.fetchVotePost).toHaveBeenCalledTimes(1)
            wrapper.find('.vote-up').simulate('click')
            expect(setupPost.fetchVotePost).toHaveBeenCalledTimes(2)
        })
    })

    describe("<VoteScore  type='comments' />", () => {
        const wrapper = shallow(<VoteScore {...setupComment} />)

        it('Should render corretly on type Comment', () => {
            expect(wrapper).toMatchSnapshot()
        })

        it('Should call fetchVoteComment when the buttons are clicked', () => {
            expect(setupComment.fetchVoteComment).toHaveBeenCalledTimes(0)
            wrapper.find('.vote-down').simulate('click')
            expect(setupComment.fetchVoteComment).toHaveBeenCalledTimes(1)
            wrapper.find('.vote-up').simulate('click')
            expect(setupComment.fetchVoteComment).toHaveBeenCalledTimes(2)
        })
    })

    it('Should not do anything when method vote is called with undefined type property', () => {
        const setup = {
            id: voteComment.id,
            score: voteComment.voteScore,
            type: 'comment',
            size: 20,
            fetchVotePost: jest.fn(),
            fetchVoteComment: jest.fn()
        }
        const wrapper = shallow(<VoteScore {...setup} />)
        expect(setup.fetchVotePost).toHaveBeenCalledTimes(0)
        expect(setup.fetchVoteComment).toHaveBeenCalledTimes(0)
        wrapper.instance().vote({ type: undefined })
        expect(setup.fetchVotePost).toHaveBeenCalledTimes(0)
        expect(setup.fetchVoteComment).toHaveBeenCalledTimes(0)
    })

})