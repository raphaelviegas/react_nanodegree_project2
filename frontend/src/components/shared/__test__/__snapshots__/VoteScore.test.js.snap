// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VoteScore /> <VoteScore  type='comments' /> Should render corretly on type Comment 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <VoteScore
    fetchVoteComment={[Function]}
    id="894tuq4ut84ut8v4t8wun89g"
    score={10}
    size={20}
    type="comment"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="vote-score-button vote-down"
          onClick={[Function]}
>
          <MaterialIcon
                    icon="thumb_down"
                    size={20}
          />
</div>,
        <span
          className="vote-score-score"
>
          10
</span>,
        <div
          className="vote-score-button vote-up"
          onClick={[Function]}
>
          <MaterialIcon
                    icon="thumb_up"
                    size={20}
          />
</div>,
      ],
      "className": "vote-score",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <MaterialIcon
            icon="thumb_down"
            size={20}
/>,
          "className": "vote-score-button vote-down",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "icon": "thumb_down",
            "size": 20,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": 10,
          "className": "vote-score-score",
        },
        "ref": null,
        "rendered": 10,
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <MaterialIcon
            icon="thumb_up"
            size={20}
/>,
          "className": "vote-score-button vote-up",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "icon": "thumb_up",
            "size": 20,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="vote-score-button vote-down"
            onClick={[Function]}
>
            <MaterialIcon
                        icon="thumb_down"
                        size={20}
            />
</div>,
          <span
            className="vote-score-score"
>
            10
</span>,
          <div
            className="vote-score-button vote-up"
            onClick={[Function]}
>
            <MaterialIcon
                        icon="thumb_up"
                        size={20}
            />
</div>,
        ],
        "className": "vote-score",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <MaterialIcon
              icon="thumb_down"
              size={20}
/>,
            "className": "vote-score-button vote-down",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "icon": "thumb_down",
              "size": 20,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 10,
            "className": "vote-score-score",
          },
          "ref": null,
          "rendered": 10,
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <MaterialIcon
              icon="thumb_up"
              size={20}
/>,
            "className": "vote-score-button vote-up",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "icon": "thumb_up",
              "size": 20,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`<VoteScore /> <VoteScore  type='posts' /> Should render corretly on type Post 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <VoteScore
    fetchVotePost={[Function]}
    id="8xf0y6ziyjabvozdd253nd"
    score={10}
    size={30}
    type="post"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="vote-score-button vote-down"
          onClick={[Function]}
>
          <MaterialIcon
                    icon="thumb_down"
                    size={30}
          />
</div>,
        <span
          className="vote-score-score"
>
          10
</span>,
        <div
          className="vote-score-button vote-up"
          onClick={[Function]}
>
          <MaterialIcon
                    icon="thumb_up"
                    size={30}
          />
</div>,
      ],
      "className": "vote-score",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <MaterialIcon
            icon="thumb_down"
            size={30}
/>,
          "className": "vote-score-button vote-down",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "icon": "thumb_down",
            "size": 30,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": 10,
          "className": "vote-score-score",
        },
        "ref": null,
        "rendered": 10,
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <MaterialIcon
            icon="thumb_up"
            size={30}
/>,
          "className": "vote-score-button vote-up",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "icon": "thumb_up",
            "size": 30,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="vote-score-button vote-down"
            onClick={[Function]}
>
            <MaterialIcon
                        icon="thumb_down"
                        size={30}
            />
</div>,
          <span
            className="vote-score-score"
>
            10
</span>,
          <div
            className="vote-score-button vote-up"
            onClick={[Function]}
>
            <MaterialIcon
                        icon="thumb_up"
                        size={30}
            />
</div>,
        ],
        "className": "vote-score",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <MaterialIcon
              icon="thumb_down"
              size={30}
/>,
            "className": "vote-score-button vote-down",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "icon": "thumb_down",
              "size": 30,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 10,
            "className": "vote-score-score",
          },
          "ref": null,
          "rendered": 10,
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <MaterialIcon
              icon="thumb_up"
              size={30}
/>,
            "className": "vote-score-button vote-up",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "icon": "thumb_up",
              "size": 30,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
